src/github.com/matrix-org/dendrite/clientapi/routing/routing.go

```diff
@@ -131,6 +131,10 @@ func Setup(
 		return writers.LegacyRegister(req, accountDB, deviceDB, &cfg)
 	})).Methods("POST", "OPTIONS")
 
+	r0mux.Handle("/register/available", common.MakeExternalAPI("registerAvailable", func(req *http.Request) util.JSONResponse {
+		return readers.RegisterAvailable(req, accountDB)
+	})).Methods("GET", "OPTIONS")
```

Comment:
Handling `OPTIONS` here isn't needed as CORS pre-flighted requests aren't sent for a `GET` request: https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests